sequence_processing_pipeline/Pipeline.py

```diff
@@ -498,47 +495,49 @@ def _validate_mapping_file(mapping_file_path):
 
         try:
             df = pd.read_csv(mapping_file_path, delimiter='\t', dtype=str)
-
-            columns = [x.lower() for x in list(df.columns)]
-            if len(set(columns)) < len(columns):
-                raise PipelineError("Column names are case-insensitive. You "
-                                    "have one or more duplicate columns in "
-                                    "your mapping-file.")
-
-            # rename all columns to their lower-case versions.
-            # we will want to return this version to the user.
-            df.columns = df.columns.str.lower()
-
-            # if the two sets of headers are equal, verify there are no
-            # duplicate sample-names and then return the dataframe and no
-            # error message.
-            if len(exp - set(df.columns)) == 0:
-                # count the number of occurances of each sample-name.
-                dupes = df['sample_name'].value_counts()
-                # filter for duplicate sample-names
-                dupes = dupes.loc[lambda x: x > 1]
-                dupes = dupes.index.tolist()
-
-                if dupes:
-                    msg = ('duplicate sample-names detected: '
-                           '%s' % ', '.join(dupes))
-                else:
-                    return df
-            else:
-                if exp - set(df.columns) == exp:
-                    # since pandas.read_csv() will successfully open a sample-
-                    # sheet, if none of the expected columns exist then assume
-                    # that this file is not a mapping file.
-                    msg = "not a mapping-file"
-                else:
-                    msg = 'missing columns: %s' % ', '.join(
-                        exp - set(df.columns))
         except pd.errors.ParserError:
             # ignore parser errors as they obviously prove this is not a
-            # valid mapping file.
-            msg = 'could not parse file "%s"' % mapping_file_path
-
-        raise PipelineError(msg)
+            # valid mapping file. Return None w/error message
+            return None, 'Could not parse file "%s"' % mapping_file_path
+
+        columns = [x.lower() for x in list(df.columns)]
+        if len(set(columns)) < len(columns):
+            # return an invalid mapping file w/error message
+            return df, ("Column names are case-insensitive. You "
+                        "have one or more duplicate columns in "
+                        "your mapping-file.")
+
+        # rename all columns to their lower-case versions.
+        # we will want to return this version to the user.
+        df.columns = df.columns.str.lower()
+
+        # if the two sets of headers are equal, verify there are no
+        # duplicate sample-names and then return the dataframe and no
+        # error message.
+        if len(exp - set(df.columns)) == 0:
```

Comment:
```suggestion
        if exp == set(df.columns):
```

...it wouldn't hurt to cast `exp` to `frozenset` at definition too

Reply:
Actually I think it might be better to redefine exp based on the consts in metapool's prep.py. Redefining the set here seems like a hack.  I'll take a look at that.